bitnami/thanos/templates/receive/service.yaml

```diff
@@ -58,6 +58,10 @@ spec:
     {{- include "common.tplvalues.render" (dict "value" .Values.receive.service.labelSelectorsOverride "context" $) | nindent 4 }}
     {{- else }}
     {{- include "common.labels.matchLabels" . | nindent 4 }}
+    {{- if eq .Values.receive.mode "dual-mode"}}
+    app.kubernetes.io/component: receive-distributor
+    {{ else }}
     app.kubernetes.io/component: receive
+    {{ end }}
```

Comment:
Adding this here, when the `receive.mode` is `dual-mode`, there will not be any service to connect to the receive stateful set pod. Is that expected? Maybe a new service for the distributor would have more sense?

Reply:
The reasoning behind that was because the distributor handles all routing to the receive pods, and it leverages the headless service for that. Functionally a service hitting the stateful set doesn't do much if the distributor is enabled, and I felt it would be more confusing to have another service (esp. if both services would have a remote_write port)